doc/dev/libraries.md

```diff
@@ -0,0 +1,84 @@
+
+- `libelektra-highlevel-c`:
+  The C high-level API for reading/writing configuration with Elektra.
+  Intended for use by applications.
+  Partially intended for use with code-generation.
+  Specifically targets C and not intended for use via bindings.
+
+  **Prefix:** `elektraHlc*` (better suggestions welcome)
+  (Note: Even though there isn't any other `highlevel` implementation, I added the `-c` suffix to show this lib is only for C)
+
+- `libelektra-opts`:
+  The API for parsing command-line arguments according to Elektra's spec.
+  **Prefix:** `elektraOpts*`
+- `libelektra-notification`:
+  The API for setting up callbacks and automatically updated variables linked to changes in the KDB.
+
+  **Prefix:** `elektraNotification*`
+
+- `libelektra-io`:
+  Elektra's API for asynchronous operations.
+
+  **Prefix:** `elektraIo*`
+
+- `libelektra-merge`:
+  The API for merging and/or detecting conflicts between two keysets.
+
+  **Prefix:** `elektraMerge`
+
+- `libelektra-pluginbase`:
+  The API _used by_ plugins.
+
+  **Prefix:** `elektraPlugin*`
+  (Note: previously `libelektra-plugin`)
+
+- `libelektra-plugin`:
+  The API for _interacting with_ plugins.
+  This includes loading plugins and calling exported functions.
+
+  **Prefix:**: `elektraPlugin*`
+  (Note: previously `libelektra-invoke` & `loader`)
+
+- `libelektra-mount`:
+  The API for manipulating mountpoints.
+
+  **Prefix:** `elektraMount*`
+
+- `libelektra-type`:
+  The API that defines Elektra's type system. (Alternative: The API that defines the type system used by `libelektra-highlevel-c`.)
+
+  **Names:** `elektra<TYPE>ToString` and `elektraKeyTo<TYPE>`
+  (Note: extracted from `libelektra-ease`)
+
+- `libelektra-ease`:
+  A collection of various other APIs that help when interacting with `ElektraKey` and `ElektraKeyset`.
+  Contains APIs that are not large, specific, or complex enough to justify a separate library and not minimal enough for `libelektra-core`, but can be implemented on top of the public API.
+  The APIs should not specifically target C and should be usable via bindings (if appropriate for the other language).
+
+  **Prefix:** `elektra*`
+  (Note: rest of old `libelektra-ease`, merged with `libelektra-meta` & `libelektra-utility`; cleanup needed)
+
+- `libelektra-operations`:
+  Like `libelektra-ease`, this library contains various additional `ElektraKey` and `ElektraKeyset` APIs beyond the minimal API.
+  However, these APIs specifically need access to private APIs of `libelektra-core` to be implemented (efficiently).
+  The APIs should not specifically target C and should be usable via bindings (if appropriate for the other language).
+  All APIs must abstract over unstable details of private APIs and provide a stable API, or **very clearly** state the stability restrictions.
+
+  **Prefix:** `elektra*`
+  (Note: rest of `libelektra-ease`, merged with `libelektra-meta` & `libelektra-utility`, plus some stuff from old core; cleanup needed)
```

Comment:
Any specific reason you want to get rid of `libelektra-meta`?

Did you consider to introduce `libelektra-array` for dealing with arrays?

Reply:
`libelektra-meta` doesn't really have many useful functions IMO (I'd remove some of them like `key*Comment`). I also didn't find a good way to define what it is. AFAICT it's the same as ease but limited to metadata, which isn't a good enough distinction IMO.

> Did you consider to introduce `libelektra-array` for dealing with arrays?

Could be worth considering, but we'd have to extend the current API (or at least plan to) for that to make sense. With the current 6 functions (1 of which is redundant) some one which aren't that generally useful, a separate library doesn't make much sense to me.

Reply:
Yes, probably array&meta can go to ease (if it doesn't get too bloated then).